lib/http_aws_sigv4.c

```diff
@@ -288,6 +288,10 @@ CURLcode Curl_output_aws_sigv4(struct Curl_easy *data, bool proxy)
     post_data_len = (size_t)data->set.postfieldsize;
   Curl_sha256it(sha_hash,
                 (const unsigned char *) post_data, post_data_len);
+  if(!(char *) sha_hash) {
+    goto fail;
+  }
```

Comment:
please make `Curl_sha256it` return error/success instead and check that instead of relying on the output data. Yes, that's a bigger change but is much cleaner and nicer code.